source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -607,7 +667,26 @@ Otherwise, the `Connection <#connection>`_ is marked as available.
 Clearing a Connection Pool
 --------------------------
 
-A Pool MUST have a method of clearing all `Connections <#connection>`_ when instructed. Rather than iterating through every `Connection <#connection>`_, this method should simply increment the generation of the Pool, implicitly marking all current `Connections <#connection>`_ as stale. The checkOut and checkIn algorithms will handle clearing out stale `Connections <#connection>`_. If a user is subscribed to Connection Monitoring events, a PoolClearedEvent MUST be emitted after incrementing the generation.
+A Pool MUST have a method of clearing all `Connections <#connection>`_ when
+instructed. Rather than iterating through every `Connection <#connection>`_,
+this method should simply increment the generation of the Pool, implicitly
+marking all current `Connections <#connection>`_ as stale. It should also
+transition the pool's state to "paused" to halt the creation of new connections
+until it is marked as "ready" again. The checkOut and checkIn algorithms will
+handle clearing out stale `Connections <#connection>`_. If a user is subscribed
+to Connection Monitoring events, a PoolClearedEvent MUST be emitted after
+incrementing the generation / marking the pool as "paused". If the pool is
+already "paused" when it is cleared, then the pool MUST NOT emit a PoolCleared
+event.
+
+As part of clearing the pool, the WaitQueue MUST also be cleared, meaning all
+requests in the WaitQueue MUST fail with errors indicating that the pool was
+cleared while the checkOut was being performed. The error returned as a result
```

Comment:
I updated the description of the error here to match the description of the error in `checkOut`.